public/source/storage.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * A wrapper around LocalStorage to support environments where LocalStorage is not available.
+ * Stores and retrieves items from LocalStorage if available and uses a non-persistent cache otherwise.
+ */
+var storage = {};
+try {
+    storage.getItem = localStorage.getItem;
```

Comment:
Assigning a function to another object will lose the `this` context and results in a `TypeError` upon invocation. That's why I used `bind`. 😉 
`localStorage.getItem.bind(localStorage)`